pennylane/gradients/gradient_transform.py

```diff
@@ -99,18 +99,26 @@ def assert_no_variance(measurements, transform_name):
         )
 
 
-def assert_no_tape_batching(tape, transform_name):
+def assert_no_trainable_tape_batching(tape, transform_name):
     """Check whether a tape is broadcasted and raise an error if this is the case.
 
     Args:
         tape (`~.QuantumScript`): measurements to analyze
         transform_name (str): Name of the gradient transform that queries the tape
     """
     if tape.batch_size is not None:
-        raise NotImplementedError(
-            f"Computing the gradient of broadcasted tapes with the {transform_name} "
-            "gradient transform is currently not supported. See #4462 for details."
-        )
+        trainable_batch_size = None
+        for idx in range(len(tape.trainable_params)):
+            op_batch_size = tape.get_operation(idx)[0].batch_size
+            if op_batch_size is not None:
+                trainable_batch_size = op_batch_size

```

Comment:
Couldn't we just raise the error here? Is there a reason to defer raising the error to later?

Reply:
Simplified the function.

Reply:
👍